docs/deployment/resource-providers/mesos.md

```diff
@@ -26,226 +26,242 @@ under the License.
+   to `docker` and by providing the image name through [mesos.resourcemanager.tasks.container.image.name]({% link deployment/config.md %}#mesos-resourcemanager-tasks-container-image-name).
 
-You will need to run a service like Marathon or Apache Aurora which takes care of restarting the JobManager process in case of node or process failures.
-In addition, Zookeeper needs to be configured like described in the [High Availability section of the Flink docs]({% link deployment/ha/index.md %}).
+### High Availability on Mesos
+
+You will need to run a service like Marathon or Apache Aurora which takes care of restarting the 
+JobManager process in case of node or process failures. In addition, Zookeeper needs to be configured 
+as described in the [High Availability section of the Flink docs]({% link deployment/ha/index.md %}).
 
 #### Marathon
 
-Marathon needs to be set up to launch the `bin/mesos-appmaster.sh` script.
-In particular, it should also adjust any configuration parameters for the Flink cluster.
+Marathon needs to be set up to launch the `bin/mesos-appmaster.sh` script. In particular, it should 
+also adjust any configuration parameters for the Flink cluster.
 
 Here is an example configuration for Marathon:
-
+{% highlight javascript %}
+{
+  "id": "flink",
+  "cmd": "/opt/flink-{{ site.version }}/bin/mesos-appmaster.sh -Djobmanager.rpc.address=$HOST -Dmesos.resourcemanager.framework.user=root -Dmesos.master=<mesos-master>:5050 -Dparallelism.default=2",
+  "user": "root",
+  "cpus": 2,
+  "mem": 2048,
+  "instances": 1,
+  "env": {
+    "MESOS_NATIVE_JAVA_LIBRARY": "/usr/lib/libmesos.so"
+  },
+  "healthChecks": [
     {
-        "id": "flink",
-        "cmd": "$FLINK_HOME/bin/mesos-appmaster.sh -Djobmanager.memory.process.size=1472m -Djobmanager.rpc.port=6123 -Drest.port=8081 -Dtaskmanager.memory.process.size=1024m -Dtaskmanager.numberOfTaskSlots=2 -Dparallelism.default=2 -Dmesos.resourcemanager.tasks.cpus=1",
-        "cpus": 1.0,
-        "mem": 1024
+      "protocol": "HTTP",
+      "path": "/",
+      "port": 8081,
+      "gracePeriodSeconds": 300,
+      "intervalSeconds": 60,
+      "timeoutSeconds": 20,
+      "maxConsecutiveFailures": 3
     }
+  ]
+}
+{% endhighlight %}
 
-When running Flink with Marathon, the whole Flink cluster including the job manager will be run as Mesos tasks in the Mesos cluster.
-
-### Configuration parameters
-
-For a list of Mesos specific configuration, refer to the [Mesos section]({% link deployment/config.md %}#mesos)
-of the configuration documentation.
+Flink is installed into `/opt/flink-{{ site.version }}` having `root` as the owner of the Flink 
+directory (notice that the user is used twice: once as a Marathon and another time as a Mesos 
+parameter) for the example configuration above to work.
+
+`<mesos-master>` needs to be set to the hostname or IP of Mesos' master node. `$HOST` is a Marathon 
+environment variable referring to the hostname of the machine the script is executed on.
```

Comment:
So $HOST is solving the problem? Nice :-)

Reply:
Yes, the Marathon environment variables are going to be resolved before executing the command.